tests/Shared/WorkloadTesting/BuildEnvironment.cs

```diff
@@ -166,19 +168,66 @@ public BuildEnvironment(bool useSystemDotNet = false, TestTargetFramework target
             LogRootPath = Path.Combine(AppContext.BaseDirectory, "logs");
         }
 
-        if (Directory.Exists(TestRootPath))
+        Console.WriteLine($"*** [{TargetFramework}] Using path for projects: {TestRootPath}");
+        CleanupTestRootPath();
+        if (!Directory.Exists(TestRootPath))
```

Comment:
You shouldn't need to check before. You can just call `Directory.CreateDirectory`.